sa67: enable loading U-Boot via DFU

Luflosi opened the pull request:
For compiling, the `am62x_r5_usbdfu.config` fragment is needed in addition to `kontron_sa67_r5_defconfig.`
Includes a revert of 0b154e3bf92aa7cf5b3e57f1b7b2e45a2bf56bd4.

Luflosi committed:
sa67: enable loading U-Boot via DFU

This enables DFU support in the R5 SPL and the A53 bootloader to load
the boot chain from a host PC and ease development. We can't just add
the appropriate options to kontron_sa67_r5_defconfig because of size
constraints. The am62x_r5_usbdfu.config fragment turns off some options
which are not required for DFU support to reduce the size and make it
fit again.

We can't include env/ti/k3_dfu.env in board/kontron/sa67/sa67.env
because unlike env/phytec/k3_dfu.env, it also includes the layout of the
eMMC and SPI, which doesn't align with ours. Ideally, env/ti/k3_dfu.env
would only include the 'dfu_alt_info_ram' variable and additional
variables would be defined in a separate file.

For compiling, the am62x_r5_usbdfu.config fragment is needed in addition
to kontron_sa67_r5_defconfig (first 'make kontron_sa67_r5_defconfig'
then 'make am62x_r5_usbdfu.config'). The same applies to
kontron_sa67_a53_defconfig and am62x_a53_usbdfu.config.

Includes a revert of 0b154e3bf92aa7cf5b3e57f1b7b2e45a2bf56bd4.

Signed-off-by: Lukas Zirpel <lukas@zirpel.de>

mwalle commented on board/kontron/sa67/sa67.env:
```diff
@@ -1,3 +1,5 @@
+#include <env/ti/k3_dfu.env>
```
This also includes the layout of the eMMC, which doesn't align with ours. Just duplicate that file and drop the all the environment variables except for the dfu_alt_info_ram. (see the phytec k3_dfu.env for example)

mwalle commented on configs/kontron_sa67_a53_defconfig:
```diff
@@ -40,6 +40,7 @@ CONFIG_SPL_SYS_MALLOC_SIMPLE=y
 CONFIG_SPL_SYS_MMCSD_RAW_MODE=y
 CONFIG_SYS_MMCSD_RAW_MODE_U_BOOT_SECTOR=0x1000
 CONFIG_SPL_DMA=y
+CONFIG_SPL_ENV_SUPPORT=y
```
These config options are all part of the am62x_a53_usbdfu.config. Could you try to just drop any changes in this file and compile with that fragment?

mwalle reviewed (changes requested):
Overall looks good. Here are some general remarks:

(1) add you SoB (git commit -s)
(2) add sentence to the commit message what this is about. I know that's already in the header, but in general the commit just repeats that info in another way. The title is short and precise and the message gives some more info. I.e. "This enables DFU support in the R5 SPL and the A53 bootloader to load the boot chain from a host PC and ease development.". Describe the reason *why* we are using the config fragments (that is because of the size constraint. Then you can add the info how to compile/configure it in the next paragraph.

Luflosi commented:
I rebased this on top of the latest release.

Luflosi committed:
sa67: enable writing U-Boot to SPI flash via DFU

Allow using the 'dfu 0 sf' command to expose the SPI flash chip via DFU.
This makes it easier to update U-Boot on the SPI flash chip.

Without the additional variable, all dfu commands fail
with the following error message:
"dfu_alt_info" env variable not defined!

'boot.img' can be used to flash the whole SPI flash if the three U-Boot
files have already been correctly assembled into one boot.img file.

This works with the standard kontron_sa67_a53_defconfig.
The am62x_a53_usbdfu.config fragment is not needed for this to work.

Signed-off-by: Lukas Zirpel <lukas@zirpel.de>

Luflosi commented:
I added a commit to enable writing U-Boot to SPI flash via DFU.

mwalle reviewed (changes requested):
Please make a copy of the am62p_r5_usbdfu.config fragment named kontron_sa67_r5_usbdfu.config with the same content.

Otherwise it looks good.

Luflosi commented:
In person, we decided to not copy this file.

Luflosi commented:
In the second commit, I added `boot.img raw 0x0 0x400000` to expose the whole SPI flash chip as an alternative to writing U-Boot in three pieces and in the first commit I used line wrapping at 72 characters.
